### PR TITLE
Fix incorrect checkmark showing on selected friends in 'SendMessageContactActivity'

### PR DESCRIPTION
**Pull request by lucasvanhalst**
The checkmark would show up for random other friends, this has been fixed.

### CODE REVIEW

File: rogerthat/src/main/java/com/mobicage/rogerthat/FriendListAdapter.java [hunk 1]
```diff
@@ -126,72 +148,72 @@ public View getView(int position, View convertView, ViewGroup parent) {
         }
 
         Friend friend = mStore.readFriendFromCursor(cursor);
+        convertView.setTag(friend);
```
Comment (RubenMattan) [inline rogerthat/src/main/java/com/mobicage/rogerthat/FriendListAdapter.java]: This is the seconds time tag was set on convertView

File: rogerthat/src/main/java/com/mobicage/rogerthat/SendMessageContactActivity.java [hunk 2]
```diff
@@ -134,81 +127,70 @@ protected void onResume() {
 
     private void initListView() {
         T.UI();
-        mListView = (ListView) findViewById(R.id.friend_list);
+        ListView mListView = findViewById(R.id.friend_list);
```
Comment (RubenMattan) [inline rogerthat/src/main/java/com/mobicage/rogerthat/SendMessageContactActivity.java]: remove the m

File: rogerthat/src/main/java/com/mobicage/rogerthat/FriendListAdapter.java (hunk 1 above)
Comment (lucasvanhalst) [reply rogerthat/src/main/java/com/mobicage/rogerthat/FriendListAdapter.java]: that's intentional

File: rogerthat/src/main/java/com/mobicage/rogerthat/plugins/messaging/ChatInfoAddMembersActivity.java [hunk 3]
```diff
@@ -196,51 +163,31 @@ protected void onResume() {
 
     private void initListView(final List<String> disabledMembers) {
         T.UI();
-        mListView = findViewById(R.id.friend_list);
-        mListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+        ListView liewview = findViewById(R.id.friend_list);
```
Comment (RubenMattan) [inline rogerthat/src/main/java/com/mobicage/rogerthat/plugins/messaging/ChatInfoAddMembersActivity.java]: typo

File: rogerthat/src/main/java/com/mobicage/rogerthat/FriendListAdapter.java [hunk 4]
```diff
@@ -230,4 +252,28 @@ public int getCount() {
...
+
+    public boolean toggleFriendStatus(Friend friend, FriendStatus status) {
+        boolean isAdded = !mFriendsMapping.containsKey(friend.email);
+        if (isAdded) {
+            mFriendsMapping.put(friend.email, status);
+        } else {
+            if (mFriendsMapping.get(friend.email) == status) {
+                mFriendsMapping.remove(friend.email);
+            } else {
+                isAdded = false;
```
Comment (RubenMattan) [inline rogerthat/src/main/java/com/mobicage/rogerthat/FriendListAdapter.java]: else if (XXX) and remove the else part

**Pull request merged**